File: Medium/Roman To Int/Java/RomanToInt.java
import java.util.Map;
import java.util.HashMap;

public class RomanToInt {
    private static Map<Character, Integer> numerals = new HashMap<>();
    static {
        numerals.put('I', 1);
        numerals.put('V', 5);
        numerals.put('X', 10);
        numerals.put('L', 50);
        numerals.put('C', 100);
        numerals.put('D', 500);
        numerals.put('M', 1000);
    }
    public static void main(String[] args) throws Exception {
        System.out.println(romanToInt("MMCDXCIV"));
    }
    public static int romanToInt(String roman) {
        int total = 0;
        for (int i = 0; i < roman.length(); i++) {
            char numeral = roman.charAt(i);
            if (i != 0) {
                char prev_numeral = roman.charAt(i - 1);
                if (numerals.get(numeral) > numerals.get(prev_numeral)) {
                    total -= numerals.get(prev_numeral) * 2;
                }
            }
            total += numerals.get(numeral);
        }
        return total;
    }
}

File: Medium/Roman To Int/Python/romantoint.py
def roman_to_integer(roman):
  numerals = {
    'I': 1,
    'V': 5,
    'X': 10,
    'L': 50, 
    'C': 100,
    'D': 500,
    'M': 1000
  }
  total = 0
  for i in range(len(roman)):
    numeral = roman[i]
    if i != 0:
      prev_numeral = roman[i - 1]
      if numerals[numeral] > numerals[prev_numeral]:
        total -= numerals[prev_numeral] * 2
    total += numerals[roman[i]]
  return total

print(roman_to_integer("MMCDXCIV"))
    